### ISSUE TITLE
RTL2832U + R820T2 100 KHz-1.7 GHz rtlsdr_write_reg failed with -1 -4 -7

### ISSUE THREAD

**Issue by kenwawa01** [librtlsdr/librtlsdr#141]
Hi!

I've lately changed the device being used to this one: https://pl.aliexpress.com/item/32965619493.html

I use ubuntu 22.04 - tried few different kernels, the newest 6.5.0-35.

But when I use

    `rtl_test`

I get for example:

```
Found 1 device(s):
  0:  Realtek, RTL2838UHIDIR, SN: 00000001

Using device 0: Generic RTL2832U OEM
Found Rafael Micro R828D tuner
rtlsdr_demod_read_reg failed with -7
Supported gain values (29): 0.0 0.9 1.4 2.7 3.7 7.7 8.7 12.5 14.4 15.7 16.6 19.7 20.7 22.9 25.4 28.0 29.7 32.8 33.8 36.4 37.2 38.6 40.2 42.1 43.4 43.9 44.5 48.0 49.6 
rtlsdr_demod_write_reg failed with -9
r82xx_write_arr: i2c wr failed=-9 reg=0a len=1
ERROR setting I2C register 0x0A to value 0F with mask 0F
Sampling at 2048000 S/s.

Info: This tool will continuously read from the device, and report if
samples get lost. If you observe no further output, everything is fine.

```
Another example:

```
Found 1 device(s):
  0:  , , SN: 

Using device 0: Generic RTL2832U OEM
rtlsdr_write_reg failed with -1
Resetting device...
rtlsdr_write_reg failed with -4
rtlsdr_write_reg failed with -4
rtlsdr_write_reg failed with -4
rtlsdr_write_reg failed with -4
rtlsdr_write_reg failed with -4
rtlsdr_demod_write_reg failed with -4
rtlsdr_demod_read_reg failed with -4
rtlsdr_demod_write_reg failed with -4
rtlsdr_demod_read_reg failed with -4
```

Another example (in different usb port but that probably doesnt count and its rather random):

```
Found 1 device(s):
  0:  Realtek, RTL2838UHIDIR, SN: 00000001

Using device 0: Generic RTL2832U OEM
Found Rafael Micro R828D tuner
r82xx_write: i2c wr failed=-1 reg=1a len=7
r82xx_write: i2c wr failed=-1 reg=0c len=1
r82xx_init: failed=-1
rtlsdr_demod_write_reg failed with -1
rtlsdr_demod_read_reg failed with -1
Supported gain values (29): 0.0 0.9 1.4 2.7 3.7 7.7 8.7 12.5 14.4 15.7 16.6 19.7 20.7 22.9 25.4 28.0 29.7 32.8 33.8 36.4 37.2 38.6 40.2 42.1 43.4 43.9 44.5 48.0 49.6 
rtlsdr_demod_write_reg failed with -1
rtlsdr_demod_read_reg failed with -1
r82xx_write: i2c wr failed=-1 reg=0a len=1
```

I tried with few different version of kernels. I updated libusb1.0 and rtlsdr libraries.

When I use ex.

`rtl_433`

I get for this

```
Found Rafael Micro R828D tuner
r82xx_write: i2c wr failed=-7 reg=0c len=1
r82xx_init: failed=-7
rtlsdr_demod_write_reg failed with -9
rtlsdr_demod_read_reg failed with -1
[SDR] Using device 0: Realtek, RTL2838UHIDIR, SN: 00000001, "Generic RTL2832U OEM"
Exact sample rate is: 250000.000414 Hz
rtlsdr_demod_write_reg failed with -9
r82xx_write: i2c wr failed=-1 reg=0a len=1
```

Any idea how to force it to work with with this device? Thanks for help and suggestions.

**Comment by kenwawa01** [comment]
Ok I solved it by myself.

First I didn't clean the old version of librtlsdrlibrary. Secondly I used this library https://github.com/steve-m/librtlsdr 2.0.2.'

When cleaning, Installing the newest master version of this https://github.com/librtlsdr/librtlsdr and updating kernel to 6.5.0-35, everything works flawlessly. Not sure if all 3 steps were necessary.

Btw, the given device at the beginning has very nice sensitivity on 433MHz. It's similar price as dvb-t/t2 tuners but the signal is much better! It might be also related with better antenna attached by default (here I'm not so sure how much its antenna effect and how much better design of this device counts...).
